tests/backends/console/arduino/test_uno.py

```diff
@@ -0,0 +1,228 @@
+"""Tests for the Arduino Uno console backend."""
+
+import pytest
+from tests.backends.console.helpers import MockConsole
+
+from j5.backends.console.arduino.uno import ArduinoUnoConsoleBackend
+from j5.components.gpio_pin import GPIOPinMode
+
+
+def test_backend_initialisation() -> None:
+    """Test that we can initialise a Backend."""
+    backend = ArduinoUnoConsoleBackend("test")
+    assert isinstance(backend, ArduinoUnoConsoleBackend)
+
+    assert len(backend._pins) == 18
+
+    # Check all DIGITAL_OUTPUT
+    assert all(pin.mode is GPIOPinMode.DIGITAL_OUTPUT for pin in backend._pins.values())
+
+    # Check all False
+    assert all(not pin.digital_state for pin in backend._pins.values())
+
+
+def test_backend_discover() -> None:
+    """
+    Test that the backend can not discover boards.
+
+    This backend does not support discovery.
+    """
+    with pytest.raises(NotImplementedError):
+        ArduinoUnoConsoleBackend.discover()
+
+
+def test_backend_firmware_version() -> None:
+    """Test that we can get the firmware version."""
+    backend = ArduinoUnoConsoleBackend("TestBoard")
+
+    assert backend.firmware_version is None
+
+
+def test_set_gpio_pin_mode() -> None:
+    """Test that we can set the mode of a GPIO pin."""
+    backend = ArduinoUnoConsoleBackend(
+        "TestBoard",
+        console_class=MockConsole,
+    )
+
+    backend._console.expects = "Set pin 10 to DIGITAL_OUTPUT"  # type: ignore
```

Comment:
I'm guessing `type: ignore` is needed here because the typechecker doesn't know that `backend._console` is a `MockConsole` rather than the more general `Console`. Could we instead pull out the console into a local variable using a cast and refer to it directly?

```python
console = cast(MockConsole, backend._console)
console.expects = ...
```

Even better, could we refactor so that the console object is created in this function and passed into the `ArduinoUnoConsoleBackend` constructor, rather than being created inside the constructor and having to be pulled out of a private attribute?

Reply:
I've copied how this is done in other `ConsoleBackend` implementations. I agree that your method is much better.

We should refactor this everywhere at once in my opinion, but that is out of scope for this PR.

Reply:
Sounds good to me.